python/lsst/resources/_resourcePath.py

```diff
@@ -418,24 +429,29 @@ def split(self) -> tuple[ResourcePath, str]:
             ResourcePath rules.
         tail : `str`
             Last path component. Tail will be empty if path ends on a
-            separator. Tail will never contain separators. It will be
-            unquoted.
+            separator or if the URI is known to be associated with a directory.
+            Tail will never contain separators. It will be unquoted.
 
         Notes
         -----
         Equivalent to `os.path.split` where head preserves the URI
         components.
         """
+        if self.dirLike:
```

Comment:
One problem remaining is this (which you can reproduce from the resources directory):

```python
>>> from lsst.resources import ResourcePath
>>> r = ResourcePath("python")
>>> r
ResourcePath("file:///Users/timj/work/lsstsw/build/resources/python")
>>> r.parent()
ResourcePath("file:///Users/timj/work/lsstsw/build/resources/")
>>> r.dirname()
ResourcePath("file:///Users/timj/work/lsstsw/build/resources/")
>>> r.isdir()
True
>>> r.dirname()
ResourcePath("file:///Users/timj/work/lsstsw/build/resources/python")
>>> r.parent()
ResourcePath("file:///Users/timj/work/lsstsw/build/resources/")
```
so `dirname()` gives the right answer for the absolute file URI only when it's allowed to check the file system to be sure and the answer changes depending on whether `isdir` has been called or not.

Reply:
You might consider removing dirname() entirely.  I just looked and outside of Butler the only thing using it is lsst.alert.packet.schema:from_uri, which only uses it for remote URLs and would get identical results from parent().  I haven't looked but I strongly suspect the Butler cases (other than the ButlerConfig thing) would be just as happy using parent() as well.

Reply:
From my point of view dirname is just "parent, but confusing."

Reply:
`os.path.dirname` is clearly ridiculous given it changes what it does based on whether it thinks it's a directory or not. `Path.parent` doesn't care.